### PR TITLE
Initial version of reusable workflow for trigger-gitlab-pipeline

### PR DESCRIPTION
**Pull request by Jauler**
Introduce reusable workflow for triggering gitlab, such that later we could enable workflow pinning.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -115,15 +115,32 @@ async function execute (inputConfig, githubConfig) {
...
+}
+
+function getBooleanEnv (name) {
+    const trueValue = ['true', 'True', 'TRUE'];
+    const falseValue = ['false', 'False', 'FALSE'];
+    const val = getEnv(name);
+    if (trueValue.includes(val)) return true;
+    if (falseValue.includes(val)) return false;
+    throw new TypeError(`Input could not be converted to boolean value: ${name}`);
+}
```
Comment (LukasPukenis) [inline index.js:133]: Slightly more generic way of doing the same
```
function getBooleanEnv(name) {
  const val = getEnv(name).toLowerCase();
  if (val === 'true') return true;
  if (val === 'false') return false;
  throw new TypeError(`Input could not be converted to boolean value: ${name}`);
}
```

File: .github/workflows/trigger-gitlab-pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      trigger-token:
+        description: 'GitLab API trigger token'
+        required: true
+      project-id:
+        description: 'GitLab project ID'
+        required: true
+
+jobs:
+  authorize-and-resolve-workflow-ref:
+    environment:  ${{ (github.event_name == 'pull_request_target' &&
```
Comment (LukasPukenis) [inline .github/workflows/trigger-gitlab-pipeline.yml:35]: Does this mean each project integrating this must have `External` and `Internal` environments set up in github?

Comment (LukasPukenis) [reply .github/workflows/trigger-gitlab-pipeline.yml:35]: Please document those in changelog, readme or commit message

File: index.js (hunk 1 above)
Comment (Jauler) [reply index.js:133]: :thinking: These two variants are not equivalent to each other, as `.toLowerCase()` would accept invalid onces (e.g. `FaLsE`), while the first one would not. And those are invalid yaml values.

File: .github/workflows/trigger-gitlab-pipeline.yml (hunk 2 above)
Comment (Jauler) [reply .github/workflows/trigger-gitlab-pipeline.yml:35]: will do

**Review by lukjok — APPROVED** [review]

File: index.js (hunk 1 above)
Comment (LukasPukenis) [reply index.js:133]: Oh, wasn't aware that yaml has these 3 variants as options, thought it doesn't care about the case. Feel free to ignore this then

**Review by LukasPukenis — APPROVED** [review]
A+ documentation

**Pull request merged**